lib/index.js

```diff
@@ -178,10 +185,10 @@ Socket.prototype._receive = function () {
             return;
         }
     }
-    if (msg == -1) return;
+  if (msg == -1) return '';
```

Comment:
Then's let's consistently return `null` on failure.  Seems weird that sockets don't like `undefined`, but are fine with `null`.  It's a very subtle distinction to make between the two.